openupgrade_scripts/scripts/account/15.0.1.1/pre-migration.py

```diff
@@ -0,0 +1,429 @@
+        """
+        UPDATE account_tax at
+        SET country_id = p.country_id
+        FROM res_company c
+        JOIN res_partner p ON p.id = c.partner_id
+        WHERE c.id = at.company_id
+        """,
+    )
+
+
+def _fill_res_company_account_journal_payment_credit_account_id(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE res_company
+        ADD COLUMN IF NOT EXISTS account_journal_payment_credit_account_id INTEGER
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE res_company c
+        SET account_journal_payment_credit_account_id = (
+        SELECT aj.payment_credit_account_id
+        FROM account_journal aj
+        WHERE aj.company_id = c.id AND aj.type IN ('bank', 'cash')
+            AND aj.payment_credit_account_id IS NOT NULL
+        LIMIT 1)
+        """,
+    )
+
+
+def _fill_res_company_account_journal_payment_debit_account_id(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE res_company
+        ADD COLUMN IF NOT EXISTS account_journal_payment_debit_account_id INTEGER
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE res_company c
+        SET account_journal_payment_debit_account_id = (
+        SELECT aj.payment_debit_account_id
+        FROM account_journal aj
+        WHERE aj.company_id = c.id AND aj.type IN ('bank', 'cash')
+            AND aj.payment_debit_account_id IS NOT NULL
+        LIMIT 1)
+        """,
+    )
+
+
+def _fast_fill_account_payment_outstanding_account_id(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE account_payment
+        ADD COLUMN IF NOT EXISTS outstanding_account_id INTEGER""",
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE account_payment ap
+        SET outstanding_account_id = CASE
+            WHEN apml.payment_account_id IS NOT NULL
+                THEN apml.payment_account_id
+            END
+        FROM account_payment_method_line apml
+        WHERE ap.payment_method_line_id IS NOT NULL
+            AND apml.id = ap.payment_method_line_id
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE account_payment ap
+        SET outstanding_account_id = CASE
+            WHEN ap.payment_type = 'inbound'
+                AND c.account_journal_payment_debit_account_id IS NOT NULL
+                THEN c.account_journal_payment_debit_account_id
+            WHEN ap.payment_type = 'outbound'
+                AND c.account_journal_payment_credit_account_id IS NOT NULL
+                THEN c.account_journal_payment_credit_account_id
+            ELSE null
+            END
+        FROM account_move am
+        JOIN account_journal aj ON am.journal_id = aj.id
+        JOIN res_company c ON c.id = aj.company_id
+        WHERE ap.move_id = am.id AND ap.payment_method_line_id IS NULL
```

Comment:
This query seems not correct on WHERE condition
```suggestion
        UPDATE account_payment ap
        SET outstanding_account_id = CASE
            WHEN ap.payment_type = 'inbound'
                THEN c.account_journal_payment_debit_account_id
            WHEN ap.payment_type = 'outbound'
                THEN c.account_journal_payment_credit_account_id
            END
        FROM account_move am
        JOIN account_journal aj ON am.journal_id = aj.id
        JOIN res_company c ON c.id = aj.company_id
        WHERE ap.move_id = am.id
            AND ap.payment_type IN ('inbound', 'outbound')
            AND ap.outstanding_account_id IS NULL
```